src/sagemaker/jumpstart/factory/estimator.py

```diff
@@ -0,0 +1,506 @@
+    environment: Optional[Dict[str, Union[str, PipelineVariable]]] = None,
+    max_retry_attempts: Optional[Union[int, PipelineVariable]] = None,
+    source_dir: Optional[Union[str, PipelineVariable]] = None,
+    git_config: Optional[Dict[str, str]] = None,
+    container_log_level: Optional[Union[int, PipelineVariable]] = None,
+    code_location: Optional[str] = None,
+    entry_point: Optional[Union[str, PipelineVariable]] = None,
+    dependencies: Optional[List[str]] = None,
+    instance_groups: Optional[List[InstanceGroup]] = None,
+    training_repository_access_mode: Optional[Union[str, PipelineVariable]] = None,
+    training_repository_credentials_provider_arn: Optional[Union[str, PipelineVariable]] = None,
+) -> JumpStartEstimatorInitKwargs:
+    """Returns kwargs required to instantiate `sagemaker.estimator.Estimator` object."""
+
+    estimator_init_kwargs: JumpStartEstimatorInitKwargs = JumpStartEstimatorInitKwargs(
+        model_id=model_id,
+        model_version=model_version,
+        role=role,
+        region=region,
+        instance_count=instance_count,
+        instance_type=instance_type,
+        keep_alive_period_in_seconds=keep_alive_period_in_seconds,
+        volume_size=volume_size,
+        volume_kms_key=volume_kms_key,
+        max_run=max_run,
+        input_mode=input_mode,
+        output_path=output_path,
+        output_kms_key=output_kms_key,
+        base_job_name=base_job_name,
+        sagemaker_session=sagemaker_session,
+        tags=tags,
+        subnets=subnets,
+        security_group_ids=security_group_ids,
+        model_uri=model_uri,
+        model_channel_name=model_channel_name,
+        metric_definitions=metric_definitions,
+        encrypt_inter_container_traffic=encrypt_inter_container_traffic,
+        use_spot_instances=use_spot_instances,
+        max_wait=max_wait,
+        checkpoint_s3_uri=checkpoint_s3_uri,
+        checkpoint_local_path=checkpoint_local_path,
+        rules=rules,
+        debugger_hook_config=debugger_hook_config,
+        tensorboard_output_config=tensorboard_output_config,
+        enable_sagemaker_metrics=enable_sagemaker_metrics,
+        enable_network_isolation=enable_network_isolation,
+        profiler_config=profiler_config,
+        disable_profiler=disable_profiler,
+        environment=environment,
+        max_retry_attempts=max_retry_attempts,
+        source_dir=source_dir,
+        git_config=git_config,
+        hyperparameters=hyperparameters,
+        container_log_level=container_log_level,
+        code_location=code_location,
+        entry_point=entry_point,
+        dependencies=dependencies,
+        instance_groups=instance_groups,
+        training_repository_access_mode=training_repository_access_mode,
+        training_repository_credentials_provider_arn=training_repository_credentials_provider_arn,
+        image_uri=image_uri,
+    )
+
+    estimator_init_kwargs = _add_model_version_to_kwargs(estimator_init_kwargs)
```

Comment:
Couldn't we set all of these defaults in the implementation of `JumpStartEstimatorInitKwargs`? I realize that that class is located in `types.py` but it seems like we end up having a lot of boilerplate code in the factory b/c of all the fields/kwargs we have to deal with.